tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/ReportAPI.java

```diff
@@ -298,4 +309,64 @@ public ReportUpdate() {
         @Schema(description = "True if values were acceptable. False if values weren’t acceptable, but user has voted for correct ones.")
         public boolean acceptable = false;
     }
+
+    @GET
+    @Produces(MediaType.TEXT_HTML)
+    @Path("/locales/{locale}/reports/{report}.html")
+    @Operation(summary = "Get the report output")
+    @APIResponse(
+        responseCode = "200",
+        description = "report HTML",
+        content = @Content(mediaType = MediaType.TEXT_HTML))
+    @APIResponse(
+        responseCode = "404",
+        description = "Locale not found")
+    public Response getReportOutput(
+        @Parameter(required = true, example = "mt", schema = @Schema(type = SchemaType.STRING)) @PathParam("locale") String locale,
+        // Note:  @Schema(implementation = ReportId.class) did not work here. The following works.
+        @Parameter(required = true, example = "compact", schema = @Schema(type = SchemaType.STRING)) @PathParam("report") ReportId report,
+        @HeaderParam(Auth.SESSION_HEADER) String session) {
+        final CookieSession mySession = Auth.getSession(session);
+        // Here we just verify that there *is* a session
+        if (mySession == null) {
+            return Auth.noSessionResponse();
+        }
+
+        // Return the output as a stream instead of buffering in memory.
+        final SurveyMain sm = mySession.sm;
+        final CLDRLocale loc = CLDRLocale.getInstance(locale);
+        final STFactory stf = sm.getSTFactory();
+        if (!SurveyMain.getLocalesSet().contains(loc)) {
+            // No such locale
+            return Response.status(Status.NOT_FOUND).build();
+        }
+        final Chart chart = Chart.forReport(report, locale);
+
+        // TODO: We could potentially extract metadata (title, etc) from the chart
+        // object here.
+
+        return Response.ok(new StreamingOutput() {
+            @Override
+            public void write(OutputStream output) throws IOException, WebApplicationException {
+                if (chart != null) {
+                    chart.writeContents(output, stf);
+                } else {
+                    switch (report) {
+                    // "Old Three" reports
+                    case compact:
+                    case datetime:
+                    case zones:
+                        try (final Writer w = new OutputStreamWriter(output);) {
+                            SurveyAjax.generateReport("r_" + report.name(), w, sm, loc);
+                        }
+                        break;
```

Comment:
the right thing to do here is clearly to migrate these to become Charts…

Reply:
Yup